power_grid_model_c/power_grid_model_cpp/include/power_grid_model_cpp/utils.hpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_CPP_UTILS_HPP
+#define POWER_GRID_MODEL_CPP_UTILS_HPP
+
+#include "basics.hpp"
+#include "handle.hpp"
+
+#include <algorithm>
+#include <complex>
+#include <limits>
+#include <type_traits>
+
+namespace power_grid_model_cpp {
+template <typename T>
+inline typename std::enable_if<std::is_floating_point<T>::value, // NOLINT(modernize-use-constraints,
+                                                                 // modernize-type-traits)
+                               bool>::type
+is_nan(T x) {
+    return std::isnan(x);
+}
```

Comment:
Do we really need such complicated template stuff. Why not just

```suggestion
inline is_nan(double x) {
    return std::isnan(x);
}
```

Reply:
We do not have `float` type anyway. Making template is overkill.

Reply:
I will simplify then. I didn't realice that only `double` is used, hence the template.